modules/FindGLFW.cmake

```diff
@@ -39,44 +41,71 @@
 #   DEALINGS IN THE SOFTWARE.
 #
 
-find_library(GLFW_LIBRARY NAMES glfw glfw3)
+# Glfw installs cmake package config files to shared/ folder which handles dependencies
+# in case glfw is built statically.
+find_package(glfw3 CONFIG)
 
-# Include dir
-find_path(GLFW_INCLUDE_DIR
-    NAMES GLFW/glfw3.h)
+if(TARGET glfw)
+    if(NOT TARGET GLFW::GLFW)
+        # The glfw cmake config defines the imported target "glfw" we alias it to unify
+        # imported target names.
+        # TODO: Cannot create alias of imported target
+        add_library(GLFW::GLFW UNKNOWN IMPORTED)
+        get_target_property(_GLFW_IMPORTED_CONFIGURATIONS glfw IMPORTED_CONFIGURATIONS)
+        get_target_property(_GLFW_IMPORTED_LOCATION_RELEASE glfw IMPORTED_LOCATION_RELEASE)
+        get_target_property(_GLFW_IMPORTED_LOCATION_DEBUG glfw IMPORTED_LOCATION_DEBUG)
+        get_target_property(_GLFW_INTERFACE_INCLUDE_DIRECTORIES glfw INTERFACE_INCLUDE_DIRECTORIES)
+        get_target_property(_GLFW_INTERFACE_LINK_LIBRARIES glfw INTERFACE_LINK_LIBRARIES)
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args("GLFW" DEFAULT_MSG
-    GLFW_LIBRARY
-    GLFW_INCLUDE_DIR)
+        if(NOT _GLFW_INTERFACE_LINK_LIBRARIES)
+            # This is fine on systems where there are no additional libraries apart from glfw itself
+            # but prevent trying to link _GLFW_INTERFACE_LINK_LIBRARIES-NOTFOUND.lib on Windows.
+            set(_GLFW_INTERFACE_LINK_LIBRARIES "")
+        endif()
 
-if(NOT TARGET GLFW::GLFW)
-    # CMake 3.0 doesn't propagate the local target as dependency upwards
-    # the tree and then complains that GLFW::GLFW target was not found,
-    # which it shouldn't. This is reproducible with the base bootstrap
-    # project *and* when using Magnum as CMake subproject. Works with both
-    # 2.8.12 and 3.1, so I'm assuming this is a CMake 3.0 bug. The
-    # workaround is to make the target GLOBAL so it's propagated upwards
-    # the tree unconditionally. For some reason, UNKNOWN targets can't be
-    # marked as GLOBAL, so I'm  biting the bullet and saying the library is
-    # shared -- CMake 3.0 is only on Debian Jessie now and I'm assuming GLFW
-    # comes from system package, which *should be* shared. Hopefully this won't
-    # bite back in the future.
```

Comment:
Hmm. Why this is no longer here? Original commit with more details: https://github.com/mosra/magnum/commit/477bf4c9587c6f4c338438602317ed5d4020d1d8

Reply:
Probably lost during a rebase 🙈

Reply:
Fixed and rebased.